src/main/scala/io/iohk/ethereum/network/PeerManagerActor.scala

```diff
@@ -53,6 +53,20 @@ class PeerManagerActor(
 
   private type PeerMap = Map[PeerId, Peer]
 
+  implicit class ConnectedPeersOps(connectedPeers: ConnectedPeers) {
+    def outgoingConnectionDemand: Int =
+      peerConfiguration.maxOutgoingPeers - connectedPeers.outgoingPeersCount
```

Comment:
I tried `peerConfiguration.maxOutgoingPeers - connectedPeers.outgoingPeersCount - connectedPeers.outgoingPendingPeersCount` but I saw negative values, so I kept this as it was the original logic anyway.